src/components/Page/index.js

```diff
@@ -1,21 +1,80 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import { withRouter, useLocation } from "react-router-dom";
 import { clearAllBodyScrollLocks } from "body-scroll-lock";
+import Web3 from 'web3';
+
+import { ConnectWalletContext, initOnboard } from "../../utils";
+import NFTTrade from "../../contracts/NFTrade.json"
 import styles from "./Page.module.sass";
 import Header from "../Header";
 import Footer from "../Footer";
 
 const Page = ({ children }) => {
   const { pathname } = useLocation();
+  const [onboard, setOnboard] = useState(null);
+  const connectWalletContext = React.useContext(ConnectWalletContext);
+
+  useEffect(() => {
+    const onboard = initOnboard({
+      address: connectWalletContext.setAddress,
+      ens: connectWalletContext.setEns,
+      network: connectWalletContext.setNetwork,
+      balance: connectWalletContext.setBalance,
+      wallet: (wallet) => {
+        if (wallet.provider) {
+          connectWalletContext.setWallet(wallet);
+
+          // Initialize web3 object
+          const web3 = new Web3(wallet.provider);
```

Comment:
It will be better to move to React context later or create abstracted method for getting web3 object.